modules/tasks.js

```diff
@@ -0,0 +1,164 @@
+    trashIcon.classList.add('las', 'la-trash-alt', 'btn');
+    trashIcon.classList.add('hidden');
+
+    // Add event listener to handle task editing functionality
+    taskMenu.addEventListener('click', () => {
+      taskItem.classList.add('task-editing');
+      taskMenu.classList.add('hidden');
+
+      const showTrashIcon = () => {
+        trashIcon.classList.remove('hidden');
+        trashIcon.classList.add('show');
+      };
+
+      const enableTaskDescriptionEditing = () => {
+        taskDescription.contentEditable = true;
+        taskDescription.focus();
+      };
+
+      const updateTaskIndexes = () => {
+        tasks.forEach((task, index) => {
+          task.index = index + 1;
+        });
+      };
+
+      const deleteTask = () => {
+        const index = tasks.indexOf(task);
+        if (index > -1) {
+          tasks.splice(index, 1);
+          renderTasks();
+          updateTaskIndexes();
+        }
+        saveTasks();
+      };
+
+      const addTrashIconEventListener = () => {
+        trashIcon.addEventListener('click', () => {
+          if (trashIcon.classList.contains('la-trash-alt')) {
+            deleteTask();
+            renderTasks();
+          }
+        });
+      };
+
+      const disableTaskDescriptionEditing = () => {
+        taskDescription.contentEditable = false;
+      };
+
+      const hideTaskMenu = () => {
+        taskMenu.classList.add('hidden');
+      };
+
+      const showTaskMenu = () => {
+        taskMenu.classList.remove('hidden');
+        taskMenu.classList.add('show');
+      };
+
+      const addKeydownEventListener = () => {
+        taskDescription.addEventListener('keydown', (event) => {
+          if (event.key === 'Enter' || event.key === 'Escape') {
+            event.preventDefault();
+            disableTaskDescriptionEditing();
+            taskItem.classList.remove('task-editing');
+            trashIcon.classList.add('hidden');
+            showTaskMenu();
+            saveTasks();
+          }
+        });
+      };
+
+      const editTask = () => {
+        if (taskItem.classList.contains('task-editing')) {
+          showTrashIcon();
+          enableTaskDescriptionEditing();
+          addTrashIconEventListener();
+          addKeydownEventListener();
+          hideTaskMenu();
+        }
+      };
+
+      updateTaskIndexes();
+      editTask();
+    });
+    taskItem.appendChild(taskMenu);
+    taskItem.appendChild(trashIcon);
+
+    taskContainer.appendChild(taskItem);
+  });
+}
+
+export function addTask(description) {
+  const index = tasks.length + 1;
+  tasks.push({
+    description,
+    completed: false,
+    index,
+  });
+  saveTasks();
+  renderTasks();
+}
+
+const updateTaskIndexes = () => {
+  tasks.forEach((task, index) => {
+    task.index = index + 1;
+  });
+};
+
+export function clearCompletedTasks() {
+  const completedTasks = tasks.filter((task) => task.completed);
+
+  for (let i = 0; i < completedTasks.length; i += 1) {
```

Comment:
Another suggestions I can give you in the best interests of improving your code skills is better if you use forEach loop instead because forEach is a more efficient tool to save memory and avoid conflicts with your code. Thank you very much for leaving me to review your code!